src/Orc.SystemInfo/Wmi/WindowsManagementObject.cs

```diff
@@ -64,24 +62,24 @@ internal WindowsManagementObject(IWbemClassObject wmiObject)
             }
         }
 
-        public void Dispose()
+        protected override void DisposeUnmanaged()
         {
-            if (!_disposed)
+            base.DisposeUnmanaged();
+            if(_wbemClassObject is not null)
             {
                 Marshal.ReleaseComObject(_wbemClassObject);
-                _disposed = true;
             }
         }
 
         public IEnumerable<string> GetPropertyNames()
         {
-            ThrowIfDisposed();
+            CheckDisposed();
             return _wbemClassObject.GetNames();
         }
 
         public object GetValue(string propertyName)
         {
-            ThrowIfDisposed();
+            CheckDisposed();
```

Comment:
not clear method name to me, is it throwing an exception?

Reply:
it came from Catel...

Reply:
It comes from Catel and is "old". We cannot rename it at this stage.